tower/src/balance/p2c/layer.rs

```diff
@@ -8,12 +8,15 @@ use tower_layer::Layer;
 ///
 /// This construction may seem a little odd at first glance. This is not a layer that takes
 /// requests and produces responses in the traditional sense. Instead, it is more like
-/// [`MakeService`](tower::make_service::MakeService) in that it takes service _descriptors_ (see
+/// [`MakeService`](crate::make::MakeService) in that it takes service _descriptors_ (see
 /// `Target` on `MakeService`) and produces _services_. Since [`Balance`] spreads requests across a
 /// _set_ of services, the inner service should produce a [`Discover`], not just a single
-/// [`Service`], given a service descriptor.
+/// [`Service`](tower_service::Service), given a service descriptor.
 ///
 /// See the [module-level documentation](..) for details on load balancing.
+///
+/// [`Balance`]: crate::balance::p2c::service::Balance
+/// [`Discover`]: crate::discover::Discover
```

Comment:
This doc comment has a mix of inline and trailing link references. Can we try use one style consistently, please?